src/position.cpp

```diff
@@ -888,18 +888,11 @@ void Position::do_move(Move m, StateInfo& newSt, bool givesCheck) {
       st->rule50 = 0;
   }
 
-  // Update incremental scores
-  st->psq += PSQT::psq[pc][to] - PSQT::psq[pc][from];
-
-  // Set capture piece
-  st->capturedPiece = captured;
-
-  // Add gating piece
-  if (is_gating(m))
+  else if (gating_type(m) != NO_GATE) // Safe because m is not a pawn move
```

Comment:
I suppose I cheated by adding this optimisation that is not related to the move encoding. Still I doubt this accounts for all of the speed gain I saw.